homeassistant/components/here_travel_time/sensor.py

```diff
@@ -175,21 +176,29 @@ async def async_setup_platform(
         destination = None
         destination_entity_id = config[CONF_DESTINATION_ENTITY_ID]
 
-    travel_mode = config[CONF_MODE]
     traffic_mode = config[CONF_TRAFFIC_MODE]
-    route_mode = config[CONF_ROUTE_MODE]
     name = config[CONF_NAME]
-    units = config.get(CONF_UNIT_SYSTEM, hass.config.units.name)
-    arrival = config.get(CONF_ARRIVAL)
-    departure = config.get(CONF_DEPARTURE)
 
-    here_data = HERETravelTimeData(
-        here_client, travel_mode, traffic_mode, route_mode, units, arrival, departure
+    here_travel_time_config = HERETravelTimeConfig(
+        origin=origin,
+        destination=destination,
+        origin_entity_id=origin_entity_id,
+        destination_entity_id=destination_entity_id,
+        travel_mode=config[CONF_MODE],
+        route_mode=config[CONF_ROUTE_MODE],
+        units=config.get(CONF_UNIT_SYSTEM, hass.config.units.name),
+        arrival=config.get(CONF_ARRIVAL),
+        departure=config.get(CONF_DEPARTURE),
     )
 
-    sensor = HERETravelTimeSensor(
-        name, origin, destination, origin_entity_id, destination_entity_id, here_data
+    coordinator = HereTravelTimeDataUpdateCoordinator(
+        hass,
+        here_client,
+        here_travel_time_config,
     )
+    await coordinator.async_config_entry_first_refresh()
+
+    sensor = HERETravelTimeSensor(name, traffic_mode, coordinator)
```

Comment:
Passing `HERETravelTimeConfig` into the constructor here could simplify the interactions with the coordinator to just be about `data`.